File: 10 - Python/Exercícios/05 - Matrizes/Exercício 06.py
#Exercício - Acima Diagonal

ordem = int(input("Qual a ordem da matriz? "))

mat = [[0 for x in range(ordem)] for x in range(ordem)]

for i in range(ordem):
    for j in range(0, ordem):
        mat[i][j] = int(input(f"Elemento [{i},{j}]: "))

soma = 0

for i in range(ordem):
    for j in range(i+1, ordem):
        soma = soma + mat[i][j]

print("")
print("Soma dos elementos acima da diagonal principal =", soma)

File: 06 - C/Exercícios/05 - Matrizes/Exercício 06.c
//Exercício - Acima Diagonal

#include <stdio.h>

int main() {

    int i, j, ordem, soma;

    printf("Qual a ordem da matriz? ");
    scanf("%d", &ordem);

    int mat[ordem][ordem];

    for(i = 0; i < ordem; i++) {
        for(j = 0; j < ordem; j++) {
            printf("Elemento [%d, %d]: ", i, j);
            scanf("%d", &mat[i][j]);
        }
    }

    soma = 0;

    for(i = 0; i < ordem; i++) {
        for(j = i+1; j < ordem; j++) {
            soma = soma  + mat[i][j];
        }
    }

    printf("\nSoma dos elementos acima da diagonal principal = %d", soma);

    return 0;
}
